collect_app/src/test/java/org/odk/collect/android/http/OpenRosaHttpInterfaceTest.java

```diff
@@ -86,6 +90,36 @@ public void executeGetRequest_sendsAcceptsGzipHeader() throws Exception {
         assertThat(request.getHeader("Accept-Encoding"), equalTo("gzip"));
     }
 
+    @Test
+    @Ignore("OkHttpConnection not fooled into https behaviour by port 8443 - scheme needs to be https")
```

Comment:
This is s weird one: Basically the old Apache based implementation would send basic authentication credentials after a challenge if the connection port was either 443 or 8443 (standard https ports) where as the new OkHttp implementation. This means that technically you could do basic auth over plain ol' HTTP running on an HTTPS standard port.

I'm guessing from reading the [Open Rosa Authentication docs](https://docs.opendatakit.org/openrosa-authentication/) that the intention is for basic auth to only happen over HTTPS and that the no server/client setups would be relying on using basic auth over plain HTTP (hackingly) setup on port 8443? Does that sound right @ggalmazor @yanokwa?

Reply:
Hi @seadowg, just to clarify, are you saying that the OkHttp implementation will not send basic credentials if using "http" even if the connection port is 443 or 8443?

To me, this sounds like correct behaviour - see the "http" check in OkHttpConnection.setCredentialsIfNeeded.

Thanks

Reply:
Agreed that basic auth should require that the scheme is HTTPS regardless of connection port. 👍

Reply:
Yeah that's right @xsteelej. I agree with yourself and @lognaturel here!

I don't want to start changing the behavior of the original implementation to be smarter and actually check the protocol. I've been trying to work on having a slightly different "fake" https server for each each implementation's test. 

The original implementation is super easy because I can trick it with port `8443` but it's harder to get https tests working for the new implementation as it's more correct and actually needs a localhost https server running. Going to reach out to some OkHttp/MockWebServer folks for some help!